Sources/AblyChat/Room.swift

```diff
@@ -133,53 +190,45 @@ internal actor DefaultRoom<LifecycleManagerFactory: RoomLifecycleManagerFactory>
     }
 
     /// The returned dictionary is guaranteed to have an entry for each element of `features`.
-    private static func createChannelsForFeatures(_ features: [RoomFeature], roomID: String, roomOptions _: RoomOptions, realtime: RealtimeClient) -> [RoomFeature: RealtimeChannelProtocol] {
+    private static func createChannelsForFeaturesWithOptions(_ featuresWithOptions: [RoomFeatureWithOptions], roomID: String, realtime: RealtimeClient) -> [RoomFeature: RealtimeChannelProtocol] {
         // CHA-RC3a
 
         // Multiple features can share a realtime channel. We fetch each realtime channel exactly once, merging the channel options for the various features that use this channel.
 
-        let featuresGroupedByChannelName = Dictionary(grouping: features) { $0.channelNameForRoomID(roomID) }
+        let featuresGroupedByChannelName = Dictionary(grouping: featuresWithOptions) { $0.toRoomFeature.channelNameForRoomID(roomID) }
 
         let pairsOfFeatureAndChannel = featuresGroupedByChannelName.flatMap { channelName, features in
             var channelOptions = RealtimeChannelOptions()
 
             // channel setup for presence and occupancy
             for feature in features {
-                if feature == .presence {
+                if case /* let */ .presence /* (presenceOptions) */ = feature {
```

Comment:
why commented?

Reply:
We're not currently using the presence options due to the commented-out code below, so it's commented out to avoid an unused-variable warning, whilst showing how the value should be used once the code below is restored